src/cls/rgw/cls_rgw.cc

```diff
@@ -363,6 +363,10 @@ static int read_bucket_header(cls_method_context_t hctx, struct rgw_bucket_dir_h
   int rc = cls_cxx_map_read_header(hctx, &bl);
   if (rc < 0)
     return rc;
+
+  if (bl.length() == 0) {
+      return -ENOENT;
```

Comment:
@oritwas why not just return a default rgw_bucket_dir_header struct? that's what rgw_bucket_init_index() creates anyway.